src/Database/PG/Query/Pool.hs

```diff
@@ -196,14 +193,10 @@ withConn :: ( MonadIO m
             , FromPGConnErr e
             )
          => PGPool
-         -> TxMode
          -> (PGConn -> ExceptT e m a)
          -> ExceptT e m a
-withConn pool txm f =
-  catchConnErr action
-  where
-    action  = withExpiringPGconn pool $
-             \connRsrc -> runTxOnConn connRsrc txm f
+withConn pool f =
+  catchConnErr $ withExpiringPGconn pool f
```

Comment:
This is a bit scary, because it's no longer calling `runTxOnConn` / the new `asTransaction`. Could you explain the reasoning? What impact will this have on our main repo?

Reply:
Yeah this looks a bit scary. I claim it's not, but do push back!

You're right that the behaviour of `withConn` changes: The old version might better have been called, `withConnAsTransaction`; the new version no longer does the transaction wrapping.

The core of the change then is to change `runTx` from `withConnAsTransaction ... execTx` to `withConn ... asTransaction ... execTx`. And the new non-transaction-wrapping `withConn` can be used nicely in `runTx'`.

The one part that's a bit dodgy is that `withConn` does change, and it's exported. I'd like to argue that's fine though, because it's only exported for use in unit tests. graphql-engine doesn't call it (and I'm working on the general assumption that that's the only user of pg-client-hs we need to worry about...). Wish I could unexport it though, to make it clearer the change is safe in that regard...

Reply:
I think that's a fair assumption to make, and I also wish there was a way around exporting functions just for testing.